plugins/kernel_v2/config/ssh_connect.rb

```diff
@@ -2,6 +2,8 @@ module VagrantPlugins
   module Kernel_V2
     class SSHConnectConfig < Vagrant.plugin("2", :config)
       DEFAULT_SSH_CONNECT_TIMEOUT = 15
+      DEFAULT_SSH_RETRIES         = 5
+      DEFAULT_SSH_RETRY_INTERVAL  = 10
```

Comment:
This number can probably come down some. The [winrm communicator](https://github.com/hashicorp/vagrant/blob/main/plugins/communicators/winrm/config.rb#L43) uses 2.